### PR TITLE
Feature/support action name as option

### PR DESCRIPTION
**Pull request by cyberxander90**

### CODE REVIEW

File: addon/adapters/batch-request.js [hunk 1]
```diff
@@ -181,25 +192,25 @@ export default JSONAPIAdapter.extend({
     return payloadHash;
   },
 
-  _buildBatchPayload(item, actionName) {
+  _buildBatchPayload(item, actionName, httpMethod) {
     let body, url;
     const modelName = item._internalModel.modelName;
 
-    if (actionName === 'POST') {
+    if (actionName === ACTION_NAME.CRAETE) {
```
Comment (sdhull) [inline addon/adapters/batch-request.js:199]: typo

File: addon/adapters/batch-request.js [hunk 2]
```diff
@@ -9,6 +9,11 @@ inflector.uncountable('sequential');
 
 const { isPresent } = Ember;
 const { JSONAPIAdapter } = DS;
+const ACTION_NAME = {
+  CRAETE: 0,
```
Comment (sdhull) [inline addon/adapters/batch-request.js:13]: typo `CRAETE` => `CREATE`

File: addon/adapters/batch-request.js [hunk 3]
```diff
@@ -22,31 +27,31 @@ export default JSONAPIAdapter.extend({
...
   batchCreate(items, options) {
-    if (isPresent(options) && isPresent(options.skipStoreUpdate)) {
-      return this._batch(items, 'POST', options.skipStoreUpdate);
-    }
+    options = options || {};
+    let httpMethod = options.httpMethod || 'POST';
+    let skipStoreUpdate = options.skipStoreUpdate || false;
 
-    return this._batch(items, 'POST', false);
+    return this._batch(items, ACTION_NAME.CRAETE, httpMethod, skipStoreUpdate, options.getIdsFunc);
```
Comment (sdhull) [inline addon/adapters/batch-request.js:34]: typo

File: addon/adapters/batch-request.js [hunk 4]
```diff
@@ -22,31 +27,31 @@ export default JSONAPIAdapter.extend({
...
   },
 
   batchUpdate(items, options) {
-    if (isPresent(options) && isPresent(options.skipStoreUpdate)) {
-      return this._batch(items, 'PATCH', options.skipStoreUpdate);
-    }
+    options = options || {};
+    let httpMethod = options.httpMethod || 'PATCH';
+    let skipStoreUpdate = options.skipStoreUpdate || false;
```
Comment (sdhull) [inline addon/adapters/batch-request.js:40]: 🙌 honestly I want a PR with *just this*

File: addon/adapters/batch-request.js [hunk 5]
```diff
@@ -22,31 +27,31 @@ export default JSONAPIAdapter.extend({
...
 
-    return this._batch(items, 'PATCH', false);
+    return this._batch(items, ACTION_NAME.UPDATE, httpMethod, skipStoreUpdate, options.getIdsFunc);
```
Comment (sdhull) [inline addon/adapters/batch-request.js:42]: I think `options.getIdsFunc` is probably a good feature but should be in a separate PR

File: addon/adapters/batch-request.js [hunk 6]
```diff
@@ -22,31 +27,31 @@ export default JSONAPIAdapter.extend({
...
   },
 
   batchDelete(items) {
-    return this._batch(items, 'DELETE');
+    return this._batch(items, ACTION_NAME.DELETE, 'DELETE');
   },
 
-  _batch(items, actionName, skipStoreUpdate) {
+  _batch(items, actionName, httpMethod, skipStoreUpdate, getIdsFunc) {
```
Comment (sdhull) [inline addon/adapters/batch-request.js:49]: I don't like this method signature. `actionName` is actually an int from the `ACTION_NAME` hash. And passing both the code and the string (`httpMethod`) seems like duplication.

**Review by sdhull — COMMENTED** [review]
Frankly I don't know why you would need to support both `PUT` and `PATCH`.

There are some things in this PR I like and other things I dislike. Personally, I'd spit this into multiple smaller PRs. And I'd add some tests.

**Comment by jamesmenera** [comment]
@sdhull @cyberxander90 Is this pull request ready to merge? other than the `craete` typo

**Comment by sdhull** [comment]
@jamesmenera I gave my 2c in my review...  I was just trying to be helpful with a review 🤷‍♂️
